cli/dcoscli/task/main.py

```diff
@@ -161,6 +174,156 @@ def _log(follow, completed, lines, task, file_):
     return 0
 
 
+def get_nested_container_id(task):
+    """ Get the nested container id from mesos state.
+
+    :param task: task definition
+    :type task: dict
+    :return: comma separated string of nested containers
+    :rtype: string
+    """
+    assert isinstance(task, dict)
+
+    # get current task state
+    task_state = task.get('state')
+    if not task_state:
+        raise DCOSException('Invalid executor info. '
+                            'Missing field `state`')
+
+    container_ids = []
+    statuses = task.get('statuses')
+    if not statuses:
+        raise DCOSException('Invalid executor info. Missing field `statuses`')
+
+    for status in statuses:
+        if 'state' not in status:
+            raise DCOSException('Invalid executor info. Missing field `state`')
+
+        if status['state'] != task_state:
+            continue
+
+        container_status = status.get('container_status')
+        if not container_status:
+            raise DCOSException('Invalid executor info. '
+                                'Missing field `container_status`')
+
+        container_id = container_status.get('container_id')
+        if not container_id:
+            raise DCOSException('Invalid executor info. '
+                                'Missing field `container_id`')
+
+        # traverse nested container_id field
+        while True:
```

Comment:
as far as i know python is missing `until .. do`  statement. I need to get the value at least once before continue to parent, do you know a better solution? i will gladly update it